crates/vm/src/system/memory/offline.rs

```diff
@@ -306,10 +333,17 @@ impl<F: PrimeField32> OfflineMemory<F> {
         let mut prev_timestamp = None;
 
         for i in 0..size as u32 {
+            let block = self.block_data.get(&(address_space, pointer + i));
+            if block.is_none() || block.unwrap().size == 0 {
+                self.block_data.insert(
+                    &(address_space, pointer + i),
+                    Self::initial_block_data(pointer + i, self.initial_block_size),
+                );
+            }
             let block = self
                 .block_data
-                .entry((address_space, pointer + i))
-                .or_insert_with(|| Self::initial_block_data(pointer + i, self.initial_block_size));
+                .get_mut(&(address_space, pointer + i))
```

Comment:
Ideally we would do `get_mut` on line 336 so we don't need to do it here, but I guess this be problematic for the borrow checker. We should find a way to do this without so many accesses though.